scalar/date32.go

```diff
@@ -0,0 +1,117 @@
+package scalar
+
+import (
+	"encoding"
+	"fmt"
+	"math"
+	"time"
+
+	"github.com/apache/arrow/go/v13/arrow"
+)
+
+type Date32 struct {
+	Valid bool
+	Value int32 // days since unix epoch
+}
+
+func (s *Date32) IsValid() bool {
+	return s.Valid
+}
+
+func (*Date32) DataType() arrow.DataType {
+	return arrow.FixedWidthTypes.Date32
+}
+
+func (s *Date32) Equal(rhs Scalar) bool {
+	if rhs == nil {
+		return false
+	}
+	r, ok := rhs.(*Date32)
+	if !ok {
+		return false
+	}
+	return s.Valid == r.Valid && s.Value == r.Value
+}
+
+func (s *Date32) String() string {
+	if !s.Valid {
+		return "(null)"
```

Comment:
use `array.NullValueStr` as it's now used as const to be returned in `arrays` Arrow package?

Reply:
```suggestion
		return array.NullValueStr
```